src/pages/home/components/recommendedProducts/recommendedProducts.tsx

```diff
@@ -0,0 +1,43 @@
+import { useSelector } from 'react-redux';
+import styles from './recommendedProducts.module.scss';
+import { RootState } from '../../../../redux/store';
+import { useNavigate } from 'react-router-dom';
+import routes from '../../../../utils/routes';
+
+interface ProductsProps {
+  data: {
+    productImages: { url: string }[];
+    details?: string;
+    id: string;
+    category: string;
+    ratings?: number;
+    price: number;
+    bonus: number;
+    collectionId: string;
+    createdAt?: string;
+    expDate: string;
+    expiredflag: boolean;
+    name: string;
+    quantity: number;
+  };
+}
+export default function RecommendedProducts(props: ProductsProps) {
+  const { data } = props;
+  const navigate = useNavigate();
+  const token = useSelector((state: RootState) => state.token.value);
+  const handleRedirectUser = () => (token ? navigate(routes.addToCart) : navigate(routes.login));
```

Comment:
no need to use `useSelector `and define the type for it,
you may use the predefined hook `useAppSellector `  which does not require all that